### DISCUSSION TITLE
Hooks order

### DISCUSSION THREAD

**Discussion by Ghirigoro** [vitest-dev/vitest#6000] [Q&A] [answered]
I have the following set of tests below:

```ts

describe("Order", async () => {

	beforeAll(async () => {
		console.log("BEFORE ALL");
	});

	afterAll(async () => {
		console.log("AFTER ALL");
	});

	beforeEach(async () => {
		console.log("BEFORE EACH");
	});

	it("test", async () => {
		console.log("TEST");
	});
});
```
I would expect the output of this to be:
```
BEFORE ALL
BEFORE EACH
TEST
AFTER ALL
```
But instead I get:
```
BEFORE ALL
AFTER ALL
BEFORE EACH
TEST
```

The docs for `afterAll` say:

> a callback to be called once after all tests have run in the current context.

But that's not the behavior I'm seeing.

I tried to change the ordering of hooks to "stack" in my vitest.config.ts file but the behavior is unchanged.

So my question is: What do I need to do to get Vitest to exhibit the expected behavior? Am I misunderstanding what the "context" is that the docs are referring to? I assume that it's the `describe` block the hooks are defined in but maybe that's wrong?

**Comment by sheremet-va** [comment, marked as answer]
The logging is batched so in your case it was just reported too fast. If you look at https://stackblitz.com/edit/vitest-dev-vitest-w3ovzg?file=test%2Fsuite.test.ts&initialPath=__vitest__/ you can see that `afterAll` is called after the test has finished running

**Reply by sheremet-va** [reply]
You can also disable console interception to validate your use case with `console.logs`: https://stackblitz.com/edit/vitest-dev-vitest-w3ovzg?file=test%2Fsuite.test.ts,vite.config.ts&initialPath=__vitest__/
